### ISSUE TITLE
"JavaScript heap out of memory" when continuously running too many scenarios

### ISSUE THREAD

**Issue by wallybb000** [serenity-js/serenity-js#550]
I encountered the problem when migrating to 2.0. 

After tracing code in Serenity/console-reporter, 
I found that [ActivityRelatedArtifacts ](https://github.com/serenity-js/serenity-js/blob/master/packages/console-reporter/src/stage/crew/console-reporter/ConsoleReporter.ts#L409)recorded too many Artifact data when continuously running those scenarios, and those data would not be delete after each scenario finished, which probably occupy too much heap and cause allocation fail.

**Comment by jan-molak** [comment]
Yeah, it looks like that bit of code might be using too much memory. That's a nice catch, thanks @wallybb000! Could you please tell me a bit more about what you mean by continuously running scenarios? How exactly do you run them?
